### DISCUSSION TITLE
Feature Request - TMAPI Kernel Explorer Patcher

### DISCUSSION THREAD

**Discussion by PHTNCx64** [aldostools/webMAN-MOD#519] [Ideas]
Apparently, this was present in Cobra, but was commented out (I don't know where exactly, Jo-Milk has told me this). Jo-Milks Developer Tool included this patch in one of his tools. (https://www.youtube.com/watch?v=iW3J5nueWo0)
Basically, poking the value **386000014E800020** in the LV2 address **8000000000003B38** will make the TMAPI Kernel Explorer show loaded modules, mutexes, semaphores, running threads and other relevant information for each running process on the PS3. Without this patch, the Kernel Explorer remains empty. Creating an option to enable, disable this would be really neat. 

To reset the kernel explorer, you need to poke **E92280087C0802A6** into **8000000000003B38** this will make it behave like it does for almost everyone that has not patched it.

**Comment by aldostools** [comment]
Thank you for your feedback. I don't use DEX or TMAPI, so I cannot test this feature.

As you mention that the patch is simple. You could try adding the following line to onxmb.bat
/poke.lv2?3B38=386000014E800020

Or try to execute it directly from the web browser
http://127.0.0.1/poke.lv2?3B38=386000014E800020

**Comment by aldostools** [comment]
I was investigating a bit more about this patch. It was introduced in 4.81 REX for DEX version of Cobra 7.53 but it was commented out with this message "Currently disabled to fix the issue with modding tools".

The patch was re-enabled by default in 4.82 REX in DEX version of Cobra 7.55.

```c
In main.c
#ifdef DEBUG
	// Patch is_debuggable to allow VSH to be attached on Target Manager
	{ vsh_patch, LI(R3, 1) },
	{ vsh_patch + 4, BLR },
#endif
```
Then it was disabled again in 4.84 REX in DEX version of Cobra 8.10. But enabled in the *debug* release of Cobra 8.10 for CEX & DEX.
Therefore to use this feature, you should install the *debug* version of Cobra 8.10 or patch LV2 manually with the poke provided above.

The Lite editions of Rebug 4.85.1 and 4.86.1 don't include this patch.
